Features rfc7231 methods

nyamsprod opened the pull request:
I've added the following methods to make the class even more useful:
- `Http::isInformational` tell whether the submitted code is an informational code
- `Http::isSuccessful` tell whether the submitted code is a successful code 
- `Http::isRedirection` tell whether the submitted code is a redirection code
- `Http::isClientError` tell whether the submitted code is a client error code
- `Http::isServerError` tell whether the submitted code is a server error code 
- `Http::isUnassigned` tell whether the submitted code is a unassigned code by IANA
- `Http::isCustom` tell whether the submitted code is considered custom in regard to RFC7231

Of course when used the developer is encourage to mix the response to have a better understanding of the status code. Hope you like the addition.

I have updated the Unit test accordingly

nyamsprod committed:
Adding utilities methods from RFC7231

The class can determine the status code class and nature
according to IANA list and RFC7231 status code category

nyamsprod committed:
Merge remote-tracking branch 'upstream/master' into features-rfc7231-methods

nyamsprod committed:
Improve coding style

nyamsprod committed:
Bug fix unassigned status code range

lukasoppermann commented:
Hey @nyamsprod thanks for this nice contribution. I will review this later today.

nyamsprod committed:
code simplification

colinodell commented on src/Httpstatus.php:
```diff
@@ -157,15 +184,152 @@ public function __construct($statusArray = [])
+     */
+    public function isCustom($code)
+    {
+        return 600 <= $this->filterHttpStatusCode($code);
```
Status codes above 5xx aren't valid according to the RFC (see #8)

colinodell commented on src/Httpstatus.php:
```diff
@@ -157,15 +184,152 @@ public function __construct($statusArray = [])
+     */
+    public function isUnused($code)
+    {
+        return 306 === $this->filterHttpStatusCode($code);
```
I'm not sure I like having a single method checking for a specific status code.  We don't have methods like `isOK()` or `isTemporaryRedirect()`, so I'd vote against something like this.

nyamsprod commented:
@colinodell I could restrict/remove the isCustom method ... the goal was to tell the end user if the status code is informational/sucessful/etc... I'm all about restricting the valid range of status code. 

This mean that we can reduce the valid range from 100 -> 999 (now) to 100 -> 599 (if this is what you suggested). That way the isCustom method would be automatically removed

The isUnused method on the other hand is based on IANA status code list

lukasoppermann commented:
I am not sure of how much value the is unused method is, as I would get the
same result using hasStatusCode.

nyamsprod commented:
Ok so let's strip the isCustom and isUnused methods then :+1:

nyamsprod committed:
Remove isCustom and isUnused methods

- Remove isCustom and isUnused methods
- The allowed status code range is now conform to RFC7231
